Pasfmt.Cursors.pas

```diff
@@ -0,0 +1,339 @@
+  SourceBreakpoint.Enabled := Self.FEnabled;
+  SourceBreakpoint.LineNumber := Self.FLineNumber;
+  SourceBreakpoint.Expression := Self.FExpression;
+  SourceBreakpoint.PassCount := Self.FPassCount;
+  SourceBreakpoint.ThreadCondition := Self.FThreadCondition;
+end;
+
+//______________________________________________________________________________________________________________________
+
+destructor TCursors.Destroy;
+begin
+  FreeAndNil(FBookmarkOffsets);
+  FreeAndNil(FBookmarkIds);
+  FreeAndNil(FBreakpointOffsets);
+  FreeAndNil(FBreakpointData);
+  inherited;
+end;
+
+//______________________________________________________________________________________________________________________
+
+constructor TCursors.Create(Buffer: IOTAEditBuffer);
+var
+  I: Integer;
+  EditView: IOTAEditView;
+  EditPos: TOTAEditPos;
+  CharPos: TOTACharPos;
+  Bookmark: Integer;
+  DebuggerServices: IOTADebuggerServices;
+  Breakpoint: IOTASourceBreakpoint;
+begin
+  SetLength(FOffsets, Buffer.EditViewCount);
+  SetLength(FRowsFromTop, Buffer.EditViewCount);
+
+  FBookmarkOffsets := TList<Integer>.Create;
+  FBookmarkIds := TList<Integer>.Create;
+  FBreakpointOffsets := TList<Integer>.Create;
+  FBreakpointData := TList<TBreakpointData>.Create;
+
+  for I := 0 to Buffer.EditViewCount - 1 do begin
+    EditView := Buffer.EditViews[I];
+
+    EditPos := EditView.CursorPos;
+    EditView.ConvertPos(True, EditPos, CharPos);
+    FOffsets[I] := EditView.CharPosToPos(CharPos);
+    FRowsFromTop[I] := EditView.Position.Row - EditView.TopRow;
+  end;
+
+  if Buffer.EditViewCount <= 0 then begin
+    Log.Debug('Buffer has no edit views');
+    Exit;
+  end;
+
+  // bookmarks are shared between all edit views of a buffer
+  EditView := Buffer.EditViews[0];
+
+  for Bookmark := 0 to 19 do begin
+    CharPos := EditView.BookmarkPos[Bookmark];
+    if CharPos.Line <> 0 then begin
+      FBookmarkOffsets.Add(EditView.CharPosToPos(CharPos));
+      FBookmarkIds.Add(Bookmark);
+    end;
+  end;
+
+  if BorlandIDEServices.QueryInterface(IOTADebuggerServices, DebuggerServices) = S_OK then begin
+    for I := 0 to DebuggerServices.SourceBkptCount - 1 do begin
+      Breakpoint := DebuggerServices.SourceBkpts[I];
+      if Breakpoint.FileName <> Buffer.FileName then
+        Continue;
+
+      CharPos.Line := Breakpoint.LineNumber;
+      CharPos.CharIndex := 0;
+      FBreakpointOffsets.Add(EditView.CharPosToPos(CharPos));
+      FBreakpointData.Add(TBreakpointData.From(Breakpoint));
+    end;
+  end;
+end;
+
+//______________________________________________________________________________________________________________________
+
+function TCursors.Serialize: TArray<Integer>;
+var
+  Copied: Integer;
+  List: TList<Integer>;
+begin
+  SetLength(Result, Length(FOffsets) + FBookmarkOffsets.Count + FBreakpointOffsets.Count);
+
+  Copied := 0;
+
+  if Length(FOffsets) > 0 then
+    Move(FOffsets[0], Result[Copied], Length(FOffsets) * SizeOf(FOffsets[0]));
```

Comment:
Can we change the four `SizeOf` invocations in this file to `SizeOf(Integer)` instead of `SizeOf(<context-dependent var that happens to be integer>)`?